### DISCUSSION TITLE
Query a datetime-like string using gt and lt

### DISCUSSION THREAD

**Discussion by ChrisGV04** [prisma/prisma#14906] [Q&A] [answered]
Hi! I'm currently working on a project for a customer that has a database in SQL Server with a management tool used in Mexico called "Contpaqi" which structures the database automatically.

It has kind of an old way of doing things, but the thing I haven't been able to solve is that in one of the tables they have a column called "CTIMESTAMP" which has a string `varchar(23)` type instead of `datetime` and it contains a string with a date-like format. Here's an example: `08/05/2022 16:29:02:741`.

Unfortunately, I cannot set the field as `DateTime` on my schema, it throws an error. I need to be able to search documents that are after a given date using `gt` but if I do that, it always returns an empty array understandably. However, if I run a raw query using the following: `WHERE CTIMESTAMP > Convert(datetime, '2022-08-5')` it does return the correct documents even on the same date-like string.

Is there a way to use that `Convert(datetime, $str)` functionality on Prisma or do I have to always use raw queries for that?

**Comment by austincrim** [comment, marked as answer]
Hey @ChrisGV04 👋,

Using [`$queryRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw) is the best way forward. Prisma does not offer a built-in `Convert` function, though you could open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for one.

**Reply by ChrisGV04** [reply]
Oh, okay. Thank you @austincrim for the suggestion. Yeah, I'm currently doing it with `$queryRaw` and it works, I just would prefer if there was a built in method to do something like that, but I guess it's a rare edge-case, since I didn't find similar issues.

And sure! I'll open a feature request for it to see if there's a way to implement it or not.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
